sources/packages/backend/apps/api/src/route-controllers/application-offering-change-request/application-offering-change-request.students.controller.ts

```diff
@@ -0,0 +1,140 @@
+import {
+  ApiNotFoundResponse,
+  ApiTags,
+  ApiUnauthorizedResponse,
+} from "@nestjs/swagger";
+import { AuthorizedParties } from "../../auth/authorized-parties.enum";
+import { AllowAuthorizedParty, UserToken } from "../../auth/decorators";
+import { ApiProcessError, ClientTypeBaseRoute } from "../../types";
+import BaseController from "../BaseController";
+import {
+  ApplicationOfferingChangesAPIOutDTO,
+  UpdateApplicationOfferingChangeRequestAPIInDTO,
+} from "./models/application-offering-change-request.dto";
+import { ApplicationOfferingChangeRequestStatus } from "@sims/sims-db";
+import { StudentUserToken } from "../../auth";
+import { STUDENT_UNAUTHORIZED_FOR_APPLICATION_OFFERING_CHANGE_REQUEST } from "../../constants";
+import { ApplicationOfferingChangeRequestControllerService } from "./application-offering-change-request.controller.service";
+import { ApplicationOfferingChangeRequestService } from "../../services";
+
+/**
+ * Application offering change request controller for students client.
+ */
+@AllowAuthorizedParty(AuthorizedParties.student)
+@Controller(
+  "application-offering-change-request/:applicationOfferingChangeRequestId",
+)
+@ApiTags(`${ClientTypeBaseRoute.Student}-application-offering-change-request`)
+export class ApplicationOfferingChangeRequestStudentsController extends BaseController {
+  constructor(
+    private readonly applicationOfferingChangeRequestService: ApplicationOfferingChangeRequestService,
+    private readonly applicationOfferingChangeRequestControllerService: ApplicationOfferingChangeRequestControllerService,
+  ) {
+    super();
+  }
+
+  /**
+   * Gets the Application Offering Change Request details.
+   * @param applicationOfferingChangeRequestId the Application Offering Change Request id.
+   * @param studentUserToken student user token to authorize the user.
+   * @returns Application Offering Change Request details.
+   */
+  @Get()
+  @ApiNotFoundResponse({
+    description: "Not able to find an Application Offering Change Request.",
+  })
+  async getById(
+    @Param("applicationOfferingChangeRequestId", ParseIntPipe)
+    applicationOfferingChangeRequestId: number,
+    @UserToken()
+    studentUserToken: StudentUserToken,
+  ): Promise<ApplicationOfferingChangesAPIOutDTO> {
+    return this.applicationOfferingChangeRequestControllerService.getById(
+      applicationOfferingChangeRequestId,
+      { studentId: studentUserToken.studentId },
+    );
+  }
+
+  /**
+   * Gets the Application Offering Change Request status.
+   * @param applicationOfferingChangeRequestId the Application Offering Change Request id.
+   * @param studentUserToken student user token to authorize the user.
+   * @returns Application Offering Change Request status.
+   */
+  @Get("application-offering-change-request-status")
+  @ApiNotFoundResponse({
+    description:
+      "Not able to get the Application Offering Change Request Status.",
+  })
+  async getApplicationOfferingChangeRequestStatusById(
```

Comment:
do we need this controller, you already have the above controller returning the status. I will say, we can use the same controller

Reply:
I also felt the same initially @ann-aot .

But then looking at the service of the above API which had multiple joins (Application, location, program, student, user and notes) felt that using a seperate API for displaying tab title may not be a bad idea.

Still no strong feeling against re-using above API.

Reply:
@dheepak-aot , I got the point, but if you see the usage of 1st controller https://github.com/bcgov/SIMS/pull/2196#discussion_r1297709798, 
There are a lot of fields that are not required on the client side, I believe only 3 fields are only required, `status`, `applicationId`, and `locationName`. 
And if its refactored, the same API can be used here, to, IMO

Reply:
@ann-aot Even after refactoring the 1st controller, the `application-offering-change-request.service` call will still remain the same. We will only be making modifications to the `application-offering-change-request.controller.service`, right? Please correct me if I am missing something here.

Reply:
@sh16011993 I don't see any issue with that, And if you see Ministry Student and Institution tab, We are doing something same

Reply:
@andrepestana-aot @guru-aot suggestions?

Reply:
I believe having it as seperate controllers, will give an abstraction layer.